### PR TITLE
Nest and map blog

### LINKED ISSUE #173: Add post about map-nest C&C
Supporting material for C&C on 8th August

### PR DESCRIPTION
**Pull request by StatsRhian**
If I'd had more time, I would have written a shorter ~letter~ blogpost

Closes #173

### CODE REVIEW

File: blogs/posts/2024-08-08-map-and-nest/map-and-nest.qmd [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+knitr:
+  opts_chunk:
+    fig.height: 2.5
+    echo: true
+    dev: png
+    dev.args:
+      bg: transparent
+---
+
+I want to share a framework that I like using occasionally for data analysis. It's the nest-and-map and it's helped me countless times when I'm working with related datasets. By combining [{purrr}](https://purrr.tidyverse.org/) mapping with [{tidyr}](https://tidyr.tidyverse.org/) nesting, I can keep raw data to hand while performing over different analyses. This allows me to easily track back from a summary or plot to the original raw data, ensuring everything is linked.
```
Comment (matt-dray) [inline blogs/posts/2024-08-08-map-and-nest/map-and-nest.qmd:15]: >while performing over different analyses

This reads like you accidentally a word?

File: blogs/posts/2024-08-08-map-and-nest/map-and-nest.qmd [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+The resulting column `barchart` is again a list-column, but this time instead of containing a tibble, it holds a ggplot object. A whole ggplot in a single cell. ^[This totally blew my mind the first time I saw it 🤯.]
+
+If we want to pass two arguments to our function, we can replace `map()` with `map2()`. Here we're using `map2()` to pass the `ics_name` column to use as a title in our waffle plot.
+
+```{r}
+graphs <-
+processed_staff |>
+    dplyr::mutate(
+        waffle =  purrr::map2(raw_data, ics_name, ~plot_waffle(.x, title = .y))
```
Comment (matt-dray) [inline blogs/posts/2024-08-08-map-and-nest/map-and-nest.qmd:89]: I wonder if adding the arg names might make clearer the connection from `raw_data` and `ics_name` to `.x` and `.y` for folks who haven't done a `map2()` before?

```suggestion
        waffle =  purrr::map2(
            .x = raw_data, 
            .y = ics_name, 
            ~plot_waffle(.x, title = .y)
        )
```

File: blogs/posts/2024-08-08-map-and-nest/map-and-nest.qmd [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+group_by_ics$raw_data[[1]]
+```
+
+Next, let's apply some simple processing, say converting absolute numbers into percentages, to each of the ICSs in turn. We use `mutate()` to create a new list-column `staff_percent` and `map()` to apply the processing function to each cell in turn. ^[In this example, we actually didn't need to nest first. We could have performed the `mutate()` step on the full dataset.]
+
+```{r}
+processed_staff <-
+group_by_ics |>
+    dplyr::mutate(
+        staff_percent = purrr::map(raw_data, convert_percent)
```
Comment (matt-dray) [inline blogs/posts/2024-08-08-map-and-nest/map-and-nest.qmd:65]: To consider: maybe a footnote on how the function can be provided as the raw function name (like in this example) vs `~` (like in the code block at line 85) vs `\()` and `function()` (block at line 117).

**Review by matt-dray — APPROVED** [review]
Smashing. A great session and folks will definitely benefit from having this blog post available to refer to. Approved assuming the typos are corrected; all other comments are for your consideration.

**Review by matt-dray — APPROVED** [review]

File: blogs/posts/2024-08-08-map-and-nest/helper-functions.R [hunk 4]
```diff
@@ -0,0 +1,51 @@
+#' Helper functions for purrr-map
```
[the remark this answers is not in the document]
Comment (StatsRhian) [reply blogs/posts/2024-08-08-map-and-nest/helper-functions.R]: Great shout - thanks

File: blogs/posts/2024-08-08-map-and-nest/map-and-nest.qmd [hunk 5]
```diff
@@ -56,7 +51,26 @@ We can grab these mini datasets in the usual way and explore them interactively.
 group_by_ics$raw_data[[1]]
 ```
 
-Next, let's apply some simple processing, say converting absolute numbers into percentages, to each of the ICSs in turn. We use `mutate()` to create a new list-column `staff_percent` and `map()` to apply the processing function to each cell in turn. ^[In this example, we actually didn't need to nest first. We could have performed the `mutate()` step on the full dataset.]
+Next, let's apply some simple processing, say converting absolute numbers into percentages, to each of the ICSs in turn.
+
+<details>
```
Comment (matt-dray) [inline blogs/posts/2024-08-08-map-and-nest/map-and-nest.qmd:55]: I'll leave it for you to decide if you want to do like:

```
<details><summary>Click for function definition.</summary>
```

Comment (StatsRhian) [reply blogs/posts/2024-08-08-map-and-nest/map-and-nest.qmd:57]: Good shout again - thanks!

File: blogs/posts/2024-08-08-map-and-nest/map-and-nest.qmd (hunk 3 above)
Comment (StatsRhian) [reply blogs/posts/2024-08-08-map-and-nest/map-and-nest.qmd:80]: I've dropped the ~ and linked to the help page

(I only still do ~ out of habit. Can't teach an old dog new tricks)

**Review by matt-dray — COMMENTED** [review]
Smashing.

**Pull request merged**